### DISCUSSION TITLE
Incomplete ArrayTypes/DictionaryTypes in TypeScriptGenerator

### DISCUSSION THREAD

**Discussion by ItsClemi** [ServiceStack/Discuss#53] [Ideas]
Hi,

we're currently running into an issue where servicestack's typescript code gen doesn't translate the `IList<T>` type of our response model. When using the underlying `ICollection<T>` type, the correct output gets emitted.

As the dart-gen worked fine, I've taken a look at the source-code and found that the fields `ArrayTypes` and `DictionaryTypes` of the following generators mismatch with the dart gen:

- TypeScriptGenerator
- SwiftGenerator
- JavaGenerator
- KotlinGenerator
- PythonGenerator

For now I've fixed the issue by manually adding the missing types to the `TypeScriptGenerator` via:

```csharp
        TypeScriptGenerator.ArrayTypes.AddDistinctRange(new[] {
            "IList`1", 
            "ConcurrentQueue`1", 
            "ConcurrentStack`1", 
            "ArrayList",
        });

        TypeScriptGenerator.DictionaryTypes.AddDistinctRange(new[] {
            "ConcurrentDictionary`2",
            "SortedList`2",
            "SortedList",
            "Hashtable",
        });
```
But shouldn't the values of `ArrayTypes` and `DictionaryTypes` match in each of these generators? 

Thanks! 🙂

**Comment by mythz** [comment]
Using Interfaces or objects in DTOs is both [highly discouraged](https://stackoverflow.com/a/10759250/85785) and [not supported in Add ServiceStack Reference languages](https://docs.servicestack.net/add-servicestack-reference#no-object-or-interface-properties).

Use known concrete collections like `List<T>` instead of `IList<T>` which should never be used in DTOs.
